File: magxxic/core/signer.py
import dkim
import time

def sign_message(message_bytes, domain, selector, private_key_path):
    """
    Sign an email message with DKIM.
    """
    try:
        with open(private_key_path, 'rb') as f:
            private_key = f.read()

        # dkim.sign(message, selector, domain, privkey, ...)
        sig = dkim.sign(
            message_bytes,
            selector.encode(),
            domain.encode(),
            private_key,
            include_headers=[b'To', b'From', b'Subject']
        )
        return sig + message_bytes
    except Exception as e:
        # print(f"DKIM signing error: {e}")
        return message_bytes

File: node_sender/magxxic/core/signer.js
const fs = require('fs');

function getDkimOptions(domain, selector, privateKeyPath) {
    try {
        const privateKey = fs.readFileSync(privateKeyPath, 'utf8');
        return {
            domainName: domain,
            keySelector: selector,
            privateKey: privateKey
        };
    } catch (err) {
        return null;
    }
}

module.exports = { signMessage };
